R/tree.R

```diff
@@ -0,0 +1,314 @@
+#' Pretty tree-like object printing
+#'
+#' A cleaner and easier to read replacement for `str` for nested list-like
+#' objects
+#'
+#' @param x A tree like object (list, etc.)
+#' @param index_arraylike Should children of containers without names have
+#'   indices used as stand-in?
+#' @param max_depth How far down the tree structure should be printed. E.g. `1`
```

Comment:
Need `max_length` as well? What happens with (e.g.) a list that's 10,000 elements long?

Reply:
I had omitted this because of using recursion but I think I found a good solution using a counter variable in the calling environment.